linalg/hypre.cpp

```diff
@@ -2511,8 +2514,16 @@ void HypreSolver::Mult(const HypreParVector &b, HypreParVector &x) const
       mfem_error("HypreSolver::Mult (...) : HypreParMatrix A is missing");
       return;
    }
+
+   if (!iterative_mode)
+   {
+      x = 0.0;
+   }
+
    if (!setup_called)
    {
+      b.HostRead();
```

Comment:
Can these two lines be moved to line 2523, just before `if (!setup_called)`, to eliminate the same lines currently on 2540-2541?